src/test/kotlin/com/mparticle/kits/RoktKitTests.kt

```diff
@@ -122,6 +122,65 @@ class RoktKitTests {
         assertEquals("123", result["attr_non_string"])
     }
 
+
+    @Test
+    fun test_prepareFinalAttributes_handlesSameKeysInAttributesAndUserAttributes() {
```

Comment:
I would also add a test for if an attribute from `selectPlacement` was part of the user attribute filters to be filtered out.